### PR TITLE
[TD-2463]Added Transcipt sending logic while passing the chat to Zendesk Agent

### PR DESCRIPTION
**Pull request by raj5036**
### What do you want to achieve?
- Pass chat transcript as attachment or text once the chat is handed over to human agent in Zendesk.

### PR Checklist

- [X] I have tested it locally and all functionalities are working fine.

### How was the code tested?

- Tested locally
- Demo of chat transcripts
image

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/zendesk-chat-service.js [hunk 1]
```diff
@@ -118,6 +118,45 @@ function ZendeskChatService() {
                     console.log('zChat.sendChatMsg ', err, data);
                 }
             );
+
+            //Sending chat transcript
+            var activeConversationInfo = JSON.parse(localStorage.getItem('kommunicate')).mckActiveConversationInfo;
```
Comment (gargshantnu) [inline webplugin/js/app/zendesk-chat-service.js:123]: This data is already there in some variable. see if we can use the same.

Comment (raj5036) [reply webplugin/js/app/zendesk-chat-service.js]: Done

**Review by gargshantnu — DISMISSED** [review]

File: webplugin/js/app/zendesk-chat-service.js [hunk 2]
```diff
@@ -118,6 +118,45 @@ function ZendeskChatService() {
...
+
+            mckUtils.ajax({
+                url: KM_PLUGIN_SETTINGS.applozicBaseUrl + "/rest/ws/message/list?startIndex=0&groupId=" + activeConversationInfo.groupId,
+                type: 'get',
+                headers: {
+                    'x-authorization': window.Applozic.ALApiService.AUTH_TOKEN,
+                },
```
[the remark this answers is not in the document]
Comment (raj5036) [reply webplugin/js/app/zendesk-chat-service.js]: Discussed with @Prochnost55 about this. We decided to move the API call to kommunicate-client file.

cc @Prochnost55

File: webplugin/js/app/kommunicate-client.js [hunk 3]
```diff
@@ -179,4 +179,26 @@ Kommunicate.client = {
...
         });
     },
+
+    /**get the entire chatListDetails by groupId 
+     * @param {Object} options 
+     * @param {Number} options.groupId
+     * @param {function} callback 
+     */
+    getChatListByGroupId: function (options, callback) {
+        var { groupId } = options;
```
Comment (Prochnost55) [inline webplugin/js/app/kommunicate-client.js:189]: Will this work in IE?

Comment (Prochnost55) [reply webplugin/js/app/kommunicate-client.js:189]: Also, what will happen if groupId is undefined?

Comment (raj5036) [reply webplugin/js/app/kommunicate-client.js]: No this wouldn't have worked in IE. Changed it. If groupId is undefined then the api will return an error.

**Review by Prochnost55 — APPROVED** [review]

**Review by gargshantnu — DISMISSED** [review]

**Review by Prochnost55 — APPROVED** [review]

**Pull request merged**